test/cmdlineTests/transient_storage_layout_value_types_interleaved_with_storage/in.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.0;
+contract A {
+    uint transient x;
+    uint y;
+    int transient z;
+    int w;
+    address transient addr;
+    address d;
+    bool transient b;
+    bool c;
+}
```

Comment:
I'd also add some constants and immutables here to make sure they're not included in either layout.

Also different visibilities (`public`, `private`, etc.).

Could also be separate cases, but IMO it would be fine to keep it in one test since CLI tests are pretty verbose.